### DISCUSSION TITLE
Robot not moving in Gazebo but in Rviz

### DISCUSSION THREAD

**Discussion by vivianweli** [iRobotEducation/create3_docs#328] [Questions]
My robot is moving in Rviz appropriately based on the commands I give in the terminal, but in Gazebo, the robot does not move. What steps could I be missing? I followed the steps through on the create3_sim GitHub page. 

Is the purpose of Gazebo building a virtual environment, such as walls? What is the difference between these 2 platforms, or how could I use them to their potential? Perhaps if there is a simple explanation or links on how these platforms work, that would be helpful! 

Thanks.

**Comment by alsora** [comment]
The two platforms are very different, although at a quick look they present a similar interface.
 - Gazebo is a simulator. It completely replaces a real robot. Rather than having an actual Create 3 robot moving in your environment, you can have a simulated robot  that moves in the Gazebo scene. The `create3_sim` package allows to simulate the Create 3 robot: it will expose all the same ROS 2 interfaces that you would expect from a real robot and that you can read here https://iroboteducation.github.io/create3_docs/api/ros2/ . So for example, the `create3_sim` package will subscribe to velocity commands and it will publish the robot pose and the sensor data according to the simulated environment.
 - RViz is a visualizer. It shows in a graphical interface data coming from ROS 2 topics. RViz for example will subscribe to the robot pose topic to show it in its view. RViz does not care whether the data is coming from a real robot or from a simulated one.

I hope that this quick description clarifies your doubts about the role of the two tools; if you need more information you should have a look at their documentations: https://gazebosim.org/home and http://wiki.ros.org/rviz.

Now back to your problem.

Do you have a real Create 3 robot? If yes, you shouldn't be running Gazebo or using the `create3_sim` package together with the real robot (you are still encouraged to use the simulator for quick prototyping and testing, but be carefult to turn off your robot while doing so, or connect it to a different WiFi network).

**Reply by vivianweli** [reply]
Thanks for the reply. I don't have the Create 3 robot connected. When I send a command to publish to a topic (for example, cmd_vel), I see the robot moving in Rviz based on my command. But the robot in my Gazebo interface does not budge. The simulation is not paused either. The robot's expected behaviour in Gazebo should be the same as in Rviz right?

**Reply by alsora** [reply]
If the robot is moving in RViz, this means that the simulator is running as expected.
Rviz does not care about commanded velocities, which are read by gazebo and used to update the robot pose.

Yes, you should see the robot move also in gazebo.
It would be helpful to see the log produced by the gazebo terminal.
However, this is an issue with the gazebo gui and not really related to create 3.
You may get better support by opening an issue in the gazebo github project.
